server/app/views/questiontypes/AddressQuestionRenderer.java

```diff
@@ -109,6 +110,30 @@ protected DivTag renderInputTags(
                 validationErrors.getOrDefault(addressQuestion.getZipPath(), ImmutableSet.of()))
             .addReferenceClass(ReferenceClasses.ADDRESS_ZIP);
 
+    InputTag correctedField =
+        j2html.TagCreator.input()
+            .withType("hidden")
+            .withName(addressQuestion.getCorrectedPath().toString())
+            .withValue(addressQuestion.getCorrectedValue().orElse(""));
+
+    InputTag latitudeField =
+        j2html.TagCreator.input()
+            .withType("hidden")
+            .withName(addressQuestion.getLatitudePath().toString())
+            .withValue(addressQuestion.getLatitudeValue().orElse(0.0).toString());
+
+    InputTag longitudeField =
+        j2html.TagCreator.input()
+            .withType("hidden")
+            .withName(addressQuestion.getLongitudePath().toString())
+            .withValue(addressQuestion.getLongitudeValue().orElse(0.0).toString());
+
+    InputTag wellKnownIdField =
+        j2html.TagCreator.input()
+            .withType("hidden")
+            .withName(addressQuestion.getWellKnownIdPath().toString())
+            .withValue(addressQuestion.getWellKnownIdValue().orElse(0L).toString());
+
```

Comment:
**suggestion (blocking):** If we're going to take the approach of setting these values by storing them in the DOM, we need to protect against a malicious request where the user has altered the values. One way to do that could be to use [`CryptographicUtils.sign`](https://github.com/civiform/civiform/blob/main/server/app/services/CryptographicUtils.java#L12) to generated an HMAC for either each of these values or all of them combined and validate it on the server when the form is submitted.

The other option is to store these values server-side instead of passing them through the DOM. The HMAC approach seems likely to be less work.

Reply:
@bion Would much rather to it server side. I haven't done anything with storing session data before. Can you point me at an existing example in the app to follow?

Reply:
https://www.playframework.com/documentation/2.8.x/JavaSessionFlash

Please check that the session is encrypted automatically by Play. If it isn't, please use an HMAC to ensure data integrity before saving the values to the DB.

Reply:
I'm removing the view change part from this PR. These changes will solely cover the internals of allowing it to be saved. The view changes will be done in a separate PR.